sympy/functions/elementary/integers.py

```diff
@@ -517,6 +517,53 @@ def _eval_is_zero(self):
     def _eval_is_negative(self):
         return False
 
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):
+        from sympy.calculus.accumulationbounds import AccumBounds
+        arg = self.args[0]
+        arg0 = arg.subs(x, 0)
+        r = self.subs(x, 0)
+
+        if arg0.is_finite:
+            if arg0.is_integer or r.is_zero:
+                if cdir == 0:
+                    ndirl = arg.dir(x, cdir=-1)
+                    ndir = arg.dir(x, cdir=1)
+                    if ndir != ndirl:
+                        raise ValueError("Two sided limit of %s around 0"
+                                            "does not exist" % self)
+                    else:
+                        return r
+                else:
+                    ndir = arg.dir(x, cdir=cdir)
+                    return r + 1 if ndir.is_negative else r
+            else:
+                return r
+        elif arg0 is S.ComplexInfinity:
+            return AccumBounds(0, 1)
```

Comment:
Maybe this should be returned in real infinities case too.

Reply:
True , I hadn't thought about this but makes complete sense .